operator/iamkeycontroller/observe.go

```diff
@@ -0,0 +1,117 @@
+package iamkeycontroller
+
+import (
+	"context"
+	"fmt"
+	pipeline "github.com/ccremer/go-command-pipeline"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"github.com/vshn/provider-exoscale/operator/steps"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/url"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"strings"
+)
+
+// Observe implements managed.ExternalClient.
+func (p *IAMKeyPipeline) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	log := controllerruntime.LoggerFrom(ctx)
+	log.V(1).Info("Observing resource")
+
+	iamKey := fromManaged(mg)
+	if iamKey.Status.AtProvider.KeyID == "" {
+		// get the data generated by Create() via annotations, since in Create() we're not allowed to update the status.
+		if KeyId, exists := iamKey.Annotations[KeyIDAnnotationKey]; exists {
+			iamKey.Status.AtProvider.KeyID = KeyId
+			delete(iamKey.Annotations, KeyIDAnnotationKey)
+			if iamKey.Annotations[BucketsAnnotationKey] != "" {
+				iamKey.Status.AtProvider.Services = exoscalev1.Services{
+					SOS: exoscalev1.SOS{
+						Buckets: strings.Split(iamKey.Annotations[BucketsAnnotationKey], ","),
+					},
+				}
+			}
```

Comment:
Again, we should retrieve this from the object returned by the exoscale API, not via annotations.